draft-ietf-taps-interface.md

```diff
@@ -997,17 +1052,22 @@ and takes optional per-Message properties (see {{send-basic}}). All Send actions
 are asynchronous, and deliver events (see {{send-events}}). Sending partial
 Messages for streaming large data is also supported (see {{send-partial}}).
 
-Messages are sent on a Connection using the Send action:
+Messages are sent on a Connection or a Connection Pool using the Send action:
 
 ~~~
-Connection.Send(messageData, messageContext?, endOfMessage?)
+msgRef := Connection.Send(messageData, messageContext?, endOfMessage?)
+msgRef := ConnectionPool.Send(reqRef?, messageData, messageContext?, endOfMessage?)
```

Comment:
Exposing the distinction between a Connection and a ConnectionPool in this way seems to add complexity to the API for no benefit. Why does the application need to know a connection pool might be in use? Isn't one of the points of TAPS to allow multiplexing and pooling when possible but to provide nominal functionality when not?

The request reference is indeed useful here (if we accept that TAPS should be useful to build interfaces atop request/response semantics, which I do think is the case) to tell the TAPS system that one message should use the same low-level transport (Transport Stack Instance in Post terminology, IIRC) as another, but ISTM this could be added to every Received event and Send call as an optional parameter.